pkg/resources/data_test.go

```diff
@@ -109,3 +109,231 @@ func TestGetCSIMigrationFeatureGates(t *testing.T) {
 		})
 	}
 }
+
+func TestKubermaticAPIImage(t *testing.T) {
+	testCases := []struct {
+		name         string
+		templateData *TemplateData
+		wantAPIImage string
+	}{
+		{
+			name: "default image",
+			templateData: &TemplateData{
+				kubermaticImage: "quay.io/kubermatic/kubermatic",
+			},
+			wantAPIImage: "quay.io/kubermatic/kubermatic",
+		},
+		{
+			name: "default image with overwrite registry",
+			templateData: &TemplateData{
+				kubermaticImage:   "quay.io/kubermatic/kubermatic",
+				OverwriteRegistry: "custom-registry.kubermatic.io",
+			},
+			wantAPIImage: "custom-registry.kubermatic.io/kubermatic/kubermatic",
+		},
+		{
+			name: "custom image with 2 parts",
+			templateData: &TemplateData{
+				kubermaticImage: "kubermatic/kubermatic",
+			},
+			wantAPIImage: "docker.io/kubermatic/kubermatic",
+		},
+		{
+			name: "custom image with 2 parts with overwrite registry",
+			templateData: &TemplateData{
+				kubermaticImage:   "kubermatic/kubermatic",
+				OverwriteRegistry: "custom-registry.kubermatic.io",
+			},
+			wantAPIImage: "custom-registry.kubermatic.io/kubermatic/kubermatic",
+		},
+		{
+			name: "custom image with 1 part",
+			templateData: &TemplateData{
+				kubermaticImage: "kubermatic",
+			},
+			wantAPIImage: "docker.io/kubermatic",
+		},
```

Comment:
This case doesn't make much sense if `overwriteRegistry` is not used (since `docker.io/kubermatic` will never work). We should either handle it in the function or just ignore it.